### PR TITLE
Use ESLint 9 to lint `libs/**`

### PR DESCRIPTION
**Pull request by sndrs**
## What are you changing?

- updates the repo to the use the ESLint 9 and the new config added in #1658
- updates `@guardian/eslint-config` slightly to re-enable comments linting
- applies some fixes

## Why?

- staying up to date

## Anything else?

yeah - noticed doing this that we don't lint anything outside `libs/**`, so will add that in a further PR

### CODE REVIEW

**Comment by sndrs** [comment]
seems like there's currently a types conflict when installing `@stylistic/eslint-plugin` and `@eslint/compat`: https://github.com/eslint/rewrite/issues/104

**Comment by sndrs** [comment]
> seems like there's currently a types conflict when installing `@stylistic/eslint-plugin` and `@eslint/compat`: [eslint/rewrite#104](https://github.com/eslint/rewrite/issues/104)

fixed by #1689

File: libs/@guardian/ab-core/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/ab-core/eslint.config.js:8]: This is ignored automatically by eslint, can see the argument for writing it anyway

Comment (oliverabrahams) [inline libs/@guardian/ab-core/eslint.config.js:8]:
```suggestion
```

File: libs/@guardian/ab-react/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	...guardian.configs.react,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/ab-react/eslint.config.js:9]:
```suggestion
```

File: libs/@guardian/browserslist-config/eslint.config.mjs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	{
+		ignores: ['node_modules', '.wireit'],
```
Comment (oliverabrahams) [inline libs/@guardian/browserslist-config/eslint.config.mjs:6]:
```suggestion
		ignores: ['.wireit'],
```

File: libs/@guardian/core-web-vitals/eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/core-web-vitals/eslint.config.js:8]:
```suggestion
```

File: libs/@guardian/identity-auth-frontend/eslint.config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/identity-auth-frontend/eslint.config.js:8]:
```suggestion
```

File: libs/@guardian/identity-auth/eslint.config.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/identity-auth/eslint.config.js:8]:
```suggestion
```

File: libs/@guardian/libs/eslint.config.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import guardian from '@guardian/eslint-config';
+
+export default [
+	...guardian.configs.recommended,
+	...guardian.configs.jest,
+	{
+		ignores: [
+			'node_modules',
```
Comment (oliverabrahams) [inline libs/@guardian/libs/eslint.config.js:8]:
```suggestion
```

**Review by oliverabrahams — COMMENTED** [review]
Removed unnecessary ignore node_modules

**Comment by sndrs** [comment]
replace (accidentally) by #1696

**Pull request merged**